### PR TITLE
Convert SHACL to Turtle

### PR DESCRIPTION
**Pull request by coret**
New SHACL now Tutle based
Dataset should have IRI
Dataset only checked when not also a Website (KB NBT)

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ const client = new GraphDbClient(
     client
   );
   const crawler = new Crawler(registrationStore, datasetStore, logger);
-  const validator = await ShaclValidator.fromUrl('shacl/dataset.jsonld');
+  const validator = await ShaclValidator.fromUrl('shacl/register.shacl.ttl');
```
Comment (ddeboer) [inline src/main.ts:32]: Let’s drop shacl from the filename as it’s already obvious from the directory.

File: shacl/register.shacl.ttl [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+##
+## Schema.org
+##
+
+#
+# Class shapes
+#
+
+reg:DatacatalogShape
+	a sh:NodeShape ;
```
Comment (ddeboer) [inline shacl/register.shacl.ttl:19]: Lots of tabs while the whole repository is space-based. Please use spaces for now, I’ll push an editorconfig to the main branch later to fix any remaining formatting issues.

File: shacl/register.shacl.ttl [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+	sh:name "Datasetbeschrijving"@nl, "Datasetdescription"@en ;
+	rdfs:comment "Een datasetbeschrijving bestaat uit een elementen die de dataset beschrijft"@nl ;
+	sh:and (
+		[ sh:targetClass schema:Dataset ; ]
+		[ sh:not ( [ sh:targetClass schema:WebPage ; ]  ) ]	
+	);
+	sh:nodeKind sh:IRI ;
+	# required properties
+	sh:property reg:SchemaNameProperty, reg:SchemaDescriptionProperty, reg:SchemaCreatorProperty, reg:SchemaLicenseProperty, reg:SchemaDistributionProperty ;
```
Comment (ddeboer) [inline shacl/register.shacl.ttl:36]: Why not have the properties as blind nodes?

File: shacl/register.shacl.ttl [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+	a sh:NodeShape ;
+	sh:name "Datacatalogus"@nl, "Datacatalog"@en ;
+	rdfs:comment "Een datacatalogus bestaat uit een set van datasetbeschrijvingen"@nl ;
+	sh:targetClass schema:DataCatalog ;
+	sh:property reg:DatatalogProperty ;
+.
+
+reg:DatasetShape
+	a sh:NodeShape ;
+	sh:name "Datasetbeschrijving"@nl, "Datasetdescription"@en ;
```
Comment (ddeboer) [inline shacl/register.shacl.ttl:28]: Two words in English: ‘data catalog’ and ‘dataset description’.

File: shacl/register.shacl.ttl [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+	rdfs:comment "Een datasetbeschrijving bestaat uit een elementen die de dataset beschrijft"@nl ;
+	sh:and (
+		[ sh:targetClass schema:Dataset ; ]
+		[ sh:not ( [ sh:targetClass schema:WebPage ; ]  ) ]
```
Comment (ddeboer) [inline shacl/register.shacl.ttl:32]: :+1:

File: shacl/register.shacl.ttl [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+reg:SchemaNameProperty a sh:PropertyShape ;
+	rdfs:comment "Naam van de dataset"@nl ;
+	sh:path schema:name ;
+	sh:minCount 1 ;
+	sh:datatype xsd:string ;
+	sh:message "Een datasetbeschrijving dient een naam te bevatten"@nl, "A dataset description should have a name"@en ;
+	sh:group reg:SchemaDatasetRequiredProperties
+.
+
+reg:SchemaDescriptionProperty a sh:PropertyShape ;
```
Comment (ddeboer) [inline shacl/register.shacl.ttl:102]: You made description required again, which makes one of the test fail (and it should).

File: shacl/register.shacl.ttl (hunk 5 above)
Comment (coret) [reply shacl/register.shacl.ttl:32]: No, it doesn't work. But the thing with target selection it just doesn't target a node and skips all associated validation. Hard to notice (use shacl validate --debug to see targeted nodes).

This construct is for the KB NBT case in which a mainEntityOfPage contains a node with is a Dataset and a WebPage. I'm looking for a construct that targets Datasets but not if it's also a WebPage. The following custom target doesn't work either (no Datasets are targeted at all):
```
  sh:target [
    a sh:SPARQLTarget ;
    sh:select """ SELECT ?this WHERE { ?this a <http://schema.org/Dataset> ; MINUS { ?node a <http://schema.org/WebPage> } } """ ;
  ] ;

```

**Comment by ddeboer** [comment]
> Dataset only checked when not also a Website (KB NBT)

I had to remove that check because it didn’t work and broke the tests.

**Pull request merged**